### DISCUSSION TITLE
Usage in native applications

### DISCUSSION THREAD

**Discussion by DGollings** [electric-sql/electric#1118] [Feature requests]
We're investigating implementing Electric but have gotten stuck on how to use it in a native app (not react-native, native)

Is there an example of how to implement Electric in Kotlin and/or Swift? Is it currently possible?
If not, we could look into porting the TS client but could do with some guidance in that case, are you open to PRs?

**Comment by linear** [comment]
<p><a href="https://linear.app/electric-sql/issue/VAX-1759/usage-in-native-applications">VAX-1759 Usage in native applications</a></p>

**Comment by msfstef** [comment]
We currently only support JS/TS based frameworks (like react-native/expo) and there's a [community Dart fork](https://github.com/SkillDevs/electric_dart) for Flutter, we don't currently support purely native iOS or Android apps and don't have any plans to do so in the short term.

That being said, we ultimately want to support native apps and we're happy to collaborate on any contributions! One possible pathway would be to run the TS client inside a JS context and have a bridge/interface for communicating with the native side.

I'll be converting this issue to a feature request discussion.

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues, PRs, and discussions. We really appreciate you taking the time to start this discussion! If it's still relevant in the new system, please reopen.
